File: packages/android/src/main/java/com/kozmos/components/Avatar/Avatar.figma.kt
package com.kozmos.components.avatar

import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=83-272")
class KozmosAvatarConnect {
    @FigmaProperty(FigmaType.Text, "Image URL")
    val imageUrl: String = ""

    @FigmaProperty(FigmaType.Text, "Fallback")
    val fallback: String = "KO"

    @Composable
    fun ComponentExample() {
        KozmosAvatar(
            imageUrl = if (imageUrl.isBlank()) null else imageUrl,
            fallbackText = fallback
        )
    }
}

File: packages/ios/Sources/Components/Avatar/Avatar.figma.swift
import SwiftUI
import Figma

struct KozmosAvatarConnect: FigmaConnect {
    let component = KozmosAvatar.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=83-272"

    @FigmaString("Image URL")
    var imageURL: String = ""

    @FigmaString("Fallback")
    var fallback: String = "KO"

    var body: some View {
        KozmosAvatar(
            imageURL: URL(string: self.imageURL),
            fallbackText: self.fallback
        )
    }
}
